shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -77,52 +185,272 @@ - (BOOL)canBecomeFirstResponder {
   return YES;
 }
 
+#pragma mark - UITextInput Overrides
+
+- (id<UITextInputTokenizer>)tokenizer {
+  if (_tokenizer == nil) {
+    _tokenizer = [[UITextInputStringTokenizer alloc] initWithTextInput:self];
+  }
+  return _tokenizer;
+}
+
+- (UITextRange*)selectedTextRange {
+  return [[_selectedTextRange retain] autorelease];
+}
+
+- (void)setSelectedTextRange:(UITextRange*)selectedTextRange {
+  if (_selectedTextRange != selectedTextRange) {
+    UITextRange* oldSelectedRange = _selectedTextRange;
```

Comment:
Curious why you used this pattern instead of just releasing `_selectedTextRange` followed by the assignment.